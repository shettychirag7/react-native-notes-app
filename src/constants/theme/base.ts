import {Theme} from './type';
import {fonts} from '../fonts';

const base: Theme = {
  fontFamily: {
    titleLight: fonts.montserrat.light,
    titleRegular: fonts.montserrat.semiBold,
    titleBold: fonts.montserrat.bold,
    bodyLight: fonts.montserrat.thin,
    bodyBold: fonts.montserrat.medium,
    bodyRegular: fonts.montserrat.regular,
  },
  fontSize: {
    sm: 14,
    md: 16,
    lg: 20,
    xl: 24,
    xxl: 32,
  },
  space: {
    xs: 4,
    sm: 8,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 40,
  },
  colors: {
    primary: {
      10: '#aceef5',
      20: '#70e3f0',
      30: '#00d7ea',
      40: '#00cee6',
      50: '#00c4e1',
      60: '#00b4ce',
      70: '#009fb3',
      80: '#008b9a',
      90: '#00676c',
    },
    secondary: {
      10: '#f5f5f5',
      20: '#eeeeee',
      30: '#e0e0e0',
      40: '#bebebe',
      50: '#9f9f9f',
      60: '#767676',
      70: '#626262',
      80: '#424242',
      90: '#212121',
    },
    neutral: {
      N10: '#F7F7F7',
      N20: '#E1E1E1',
      N30: '#A8A8A8',
      N40: '#757575',
      N50: '#565656',
      N60: '#3B3B3B',
      N70: '#222222',
      N80: '#121212',
      white: '#FFFFFF',
      black: '#000000',
    },
  },
};

export default base;
