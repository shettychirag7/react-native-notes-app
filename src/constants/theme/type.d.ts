export type Theme = {
  fontFamily: {
    titleLight: string;
    titleRegular: string;
    titleBold: string;
    bodyLight: string;
    bodyBold: string;
    bodyRegular: string;
  };
  fontSize: {
    sm: number;
    md: number;
    lg: number;
    xl: number;
    xxl: number;
  };
  space: {
    xs: number;
    sm: number;
    md: number;
    lg: number;
    xl: number;
    xxl: number;
  };
  colors: {
    primary: {
      10: string;
      20: string;
      30: string;
      40: string;
      50: string;
      60: string;
      70: string;
      80: string;
      90: string;
    };
    secondary: {
      10: string;
      20: string;
      30: string;
      40: string;
      50: string;
      60: string;
      70: string;
      80: string;
      90: string;
    };
    neutral: {
      N10: string;
      N20: string;
      N30: string;
      N40: string;
      N50: string;
      N60: string;
      N70: string;
      N80: string;
      white: string;
      black: string;
    };
  };
};
