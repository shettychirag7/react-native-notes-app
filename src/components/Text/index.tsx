import React from 'react';
import {Text as RNText, TextProps as RNTextProps} from 'react-native';

import useStylePicker from './text.style';

interface TextProps extends RNTextProps {
  /**
   * type of text to render. Props is mandatory
   */
  type:
    | 'pageTitle'
    | 'sectionTitle'
    | 'cardTitle'
    | 'body'
    | 'boldBody'
    | 'bodyLight';
}

const Text = (props: TextProps) => {
  const {type, style: styleProps, ...rest} = props;

  const style = useStylePicker();

  return (
    <RNText style={[style[type], styleProps]} {...rest}>
      {props.children}
    </RNText>
  );
};

export default Text;
