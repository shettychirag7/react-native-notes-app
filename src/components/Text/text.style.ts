import {StyleSheet} from 'react-native';

import {createStylePickerHook, GeneratorProps} from '../../hooks/style';

const getStyleObject = ({theme}: GeneratorProps) =>
  StyleSheet.create({
    pageTitle: {
      fontFamily: theme.fontFamily.titleBold,
      fontSize: theme.fontSize.xl,
      letterSpacing: 0.1,
      lineHeight: theme.fontSize.xl * 1.35,
      color: theme.colors.neutral.N70,
    },
    sectionTitle: {
      fontFamily: theme.fontFamily.titleRegular,
      fontSize: theme.fontSize.lg,
      letterSpacing: 0.1,
      lineHeight: theme.fontSize.lg * 1.35,
      color: theme.colors.neutral.N70,
    },
    cardTitle: {
      fontFamily: theme.fontFamily.titleBold,
      fontSize: theme.fontSize.md,
      letterSpacing: 0.1,
      lineHeight: theme.fontSize.md * 1.35,
      color: theme.colors.neutral.N70,
    },
    body: {
      fontFamily: theme.fontFamily.bodyRegular,
      fontSize: theme.fontSize.md,
      color: theme.colors.neutral.N70,
    },
    boldBody: {
      fontFamily: theme.fontFamily.bodyBold,
      fontSize: theme.fontSize.md,
      letterSpacing: 0.1,
      color: theme.colors.neutral.N70,
    },
    bodyLight: {
      fontFamily: theme.fontFamily.bodyBold,
      fontSize: theme.fontSize.sm,
      color: theme.colors.primary[80],
    },
  });

const style = createStylePickerHook(getStyleObject);

export default style;
