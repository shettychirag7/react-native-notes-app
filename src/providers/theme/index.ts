import {createContext} from 'react';
import themes from '../../constants/theme';

export const ThemeContext = createContext({
  theme: themes.base,
});

export const ThemeProvider = ThemeContext.Provider;
