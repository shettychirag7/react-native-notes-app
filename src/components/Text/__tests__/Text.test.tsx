import React from 'react';
import {render} from '@testing-library/react-native';
import Text from '..';
import {ThemeProvider} from '../../../providers/theme';
import themes from '../../../constants/theme';

describe('Text component tests', () => {
  it('renders page title text', () => {
    const {toJSON} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <Text type={'pageTitle'}>{'Hello World'}</Text>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders section title text', () => {
    const {toJSON} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <Text type={'sectionTitle'}>{'Hello World'}</Text>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders body text', () => {
    const {toJSON} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <Text type={'body'}>{'Hello World'}</Text>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders bold body text', () => {
    const {toJSON} = render(
      <ThemeProvider value={{theme: themes.base}}>
        <Text type={'boldBody'}>{'Hello World'}</Text>
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
