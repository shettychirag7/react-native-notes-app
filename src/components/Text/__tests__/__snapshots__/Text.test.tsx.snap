// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component tests renders body text 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#222222",
        "fontFamily": "Montserrat-Regular",
        "fontSize": 16,
      },
      undefined,
    ]
  }
>
  Hello World
</Text>
`;

exports[`Text component tests renders bold body text 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#222222",
        "fontFamily": "Montserrat-Medium",
        "fontSize": 16,
        "letterSpacing": 0.1,
      },
      undefined,
    ]
  }
>
  Hello World
</Text>
`;

exports[`Text component tests renders page title text 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#222222",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 24,
        "letterSpacing": 0.1,
        "lineHeight": 32.400000000000006,
      },
      undefined,
    ]
  }
>
  Hello World
</Text>
`;

exports[`Text component tests renders section title text 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#222222",
        "fontFamily": "Montserrat-SemiBold",
        "fontSize": 20,
        "letterSpacing": 0.1,
        "lineHeight": 27,
      },
      undefined,
    ]
  }
>
  Hello World
</Text>
`;
